test/functionalTest/cases/2893_case_insensitive_custom_notif_http_headers/case_insensitive_custom_notif_http_headers.test

```diff
@@ -0,0 +1,153 @@
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Case sensitivness in httpCustom notification headers  
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for E1 with custom header names
+# 02. Create E1 with attributes A1, A2, A3 and 4
+# 03. Dump accumulator, see 1 notification
+#
+
+echo "01. Create subscription for E1 with custom header names that collide with the default ones"
+echo "=============================================================="
+payload='{
+  "subject": {
+    "entities": [
+      {
+        "id" : "E1"
+      }
+    ],
+    "condition": {
+      "attrs": []
+    }
+  },
+  "notification": {
+    "httpCustom": {
+      "url":     "http://127.0.0.1:'${LISTENER_PORT}'/notify",
+      "headers": { "${A3}": "${A4}" }
+    }
+  }
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create E1 with attributes A1, A2, A3 and A4"
+echo "==============================================="
+payload='{
+  "id": "E1",
+  "A1": "foo",
+  "A2": "bar",
+  "A3": "Content-Type" ,
+  "A4": "application/json"
```

Comment:
I'd suggest to use something that produce a header differente in regular (non-customized) notifications, i.e. "text/plain".

Reply:
Actually, we could keep case_insensitive_custom_notif_http_headers.test as it is now and have the text plain version in a different file e.g. case_insensitive_custom_notif_http_headers_with_text_plain.test).

Reply:
Do you mean to create a new test with "A4": "text/plain" and then with an expected payload like the one you describe in the comment below?

Reply:
Right

Reply:
Fixed in 5156484